### PR TITLE
Implement support to set alpha range in create_colormap and yaml colorize enhancements

### PR DESCRIPTION
**Pull request by ameraner**
This PR adds the possibility to define an alpha channel range to be applied to a colorise colormap, e.g. through the yaml enhancement configuration.

 - [x] Tests added 
 - [x] Fully documented 

With this, composites like the one below are more easily achievable, with e.g.
```yaml
  flash_area:
    standard_name: flash_area
    operations:
    - name: colorize
      method: !!python/name:satpy.enhancements.colorize
      kwargs:
        palettes:
        - {colors: ylorrd, min_value: 0, max_value: 20,
           min_alpha: 100, max_alpha: 255}
```
image

### CODE REVIEW

**Review by djhoese — CHANGES_REQUESTED** [review]
The colormap logic was actually moved to trollimage a while ago (by @gerritholl I think) but I don't think we ever migrated Satpy to use the trollimage function or required satpy to use the newer version of trollimage. I think this PR should maybe be migrated to trollimage and a separate PR be setup to switch to requiring the version of trollimage that introduced this logic. Sorry I don't have more details about what changed and what should be changed. It's been a while since I've looked at this.

https://github.com/pytroll/trollimage/blob/640fb8aca6cf384fa17721ed1ab982df459e5ac2/trollimage/colormap.py#L417-L422

**Comment by ameraner** [comment]
Hi Dave, thanks for the review! Indeed, at the beginning I thought of this only as an improved interface from satpy to the `Colorbar`, but then I also realised that this functionality is probably useful and better places in trollimage directly.

So I created a new method in trollimage: https://github.com/pytroll/trollimage/pull/170 

I kept the tests for the functionality here, and and of course they're failing now since we need the trollimage PR released first.

**Comment by ameraner** [comment]
After the released update in https://github.com/pytroll/trollimage/pull/170, and the according change in the required trollimage version, the tests related to this PR are passing again as expected, so from that point of view I believe this PR is ready. 

However, now there are failures on the `sar_c_safe` reader tests that are unrelated to this PR...

**Comment by mraspaud** [comment]
The SAR tests broke because of backwards incompatibility introduced in rioxarray release 0.15.6 and in particular this PR https://github.com/corteva/rioxarray/pull/787
I suggest blacklisting rioxarray 0.15.6 for now, until we figure a solution with @snowman2 on that PR

**Comment by ameraner** [comment]
Tests passing again after blacklisting rioxarray 0.15.6

**Comment by ameraner** [comment]
I reverted the rioxarray blacklist after Martin's fix, and tests pass again now after Dave's fixes yesterday.

So from my side this is good to go again.

File: satpy/enhancements/__init__.py [hunk 1]
```diff
@@ -455,12 +455,37 @@ def create_colormap(palette, img=None):  # noqa: D417
...
+    if palette.get("reverse", False):
+        cmap.reverse()
+    if "min_value" in palette and "max_value" in palette:
+        cmap.set_range(palette["min_value"], palette["max_value"])
+    elif "min_value" in palette or "max_value" in palette:
+        raise ValueError("Both 'min_value' and 'max_value' must be specified (or neither).")
+
+    if "min_alpha" in palette and "max_alpha" in palette:
+        cmap.set_alpha_range(palette["min_alpha"]/color_scale,
+                             palette["max_alpha"]/color_scale)
```
Comment (djhoese) [inline satpy/enhancements/__init__.py:478]:
```suggestion
        cmap.set_alpha_range(palette["min_alpha"] / color_scale,
                             palette["max_alpha"] / color_scale)
```

File: satpy/tests/enhancement_tests/test_enhancements.py [hunk 2]
```diff
@@ -350,6 +350,31 @@ def test_cmap_vrgb_as_rgba(self):
             assert cmap.values[0] == 0
             assert cmap.values[-1] == 1.0
 
+    def test_cmap_with_alpha_set(self):
+        """Test that the min_alpha and max_alpha arguments set the alpha channel correctly."""
+        with closed_named_temp_file(suffix=".npy") as cmap_filename:
+            cmap_data = _generate_cmap_test_data(None, "RGB")
+            np.save(cmap_filename, cmap_data)
+            cmap = create_colormap({"filename": cmap_filename, "min_alpha": 100, "max_alpha": 255})
```
Comment (djhoese) [inline satpy/tests/enhancement_tests/test_enhancements.py:358]: Could the code below this block be de-dented for code clarity? The temp file is only needed when `create_colormap` is called right?

File: satpy/tests/enhancement_tests/test_enhancements.py [hunk 3]
```diff
@@ -350,6 +350,31 @@ def test_cmap_vrgb_as_rgba(self):
...
+            # check that values have not been changed
+            assert cmap.values.shape[0] == 4
+            assert cmap.values[0] == 0
+            assert cmap.values[-1] == 1.0
+
+            # check that if a value is missing we raise a ValueError
+            with pytest.raises(ValueError, match="Both 'min_alpha' and 'max_alpha' must be specified*."):
+                create_colormap({"filename": cmap_filename, "max_alpha": 255})
+            with pytest.raises(ValueError, match="Both 'min_alpha' and 'max_alpha' must be specified*."):
+                create_colormap({"filename": cmap_filename, "min_alpha": 255})
```
Comment (djhoese) [inline satpy/tests/enhancement_tests/test_enhancements.py:376]: Could these be put in a separate test? Maybe a single test with parametrize for min_alpha and max_alpha.

**Review by djhoese — CHANGES_REQUESTED** [review]
Looks pretty good. Just a few small requests.

File: satpy/enhancements/__init__.py (hunk 1 above)
Comment (ameraner) [reply satpy/enhancements/__init__.py]: Done

File: satpy/tests/enhancement_tests/test_enhancements.py (hunk 2 above)
Comment (ameraner) [reply satpy/tests/enhancement_tests/test_enhancements.py:360]: Done, I've applied the same to the other similar neighbouring tests

File: satpy/tests/enhancement_tests/test_enhancements.py (hunk 3 above)
Comment (ameraner) [reply satpy/tests/enhancement_tests/test_enhancements.py]: Done

**Review by djhoese — APPROVED** [review]
Looks good to me. I'm not in love with `color_scale =` being accessed outside of `_get_cmap_from_palette_info` as that information is part of the palette, but I don't see an obvious better solution given what you need to do.

If we can get one other maintainers review that'd be great, but if no one is available in the next day or so I can merge this for you so you can continue your work.

**Pull request merged**